### PR TITLE
Automatically float xwayland windows

### PR DESCRIPTION
**Pull request by emersion**
This PR makes the compositor connect to xwayland to get fetch atoms. These atoms are then used to automatically float xwayland windows.

Reference: https://github.com/i3/i3/blob/05a1dc0f6e1b2c963b10ff5d69789f6eb54b1b68/src/manage.c#L441

Updates #2063

### CODE REVIEW

File: sway/desktop/xwayland.c [hunk 1]
```diff
@@ -199,15 +208,32 @@ static void set_fullscreen(struct sway_view *view, bool fullscreen) {
...
-	//
-	// We also want to return true if the NET_WM_STATE is MODAL.
-	// wlroots doesn't appear to provide all this information at the moment.
+	if (xwayland_view_from_view(view) == NULL) {
+		return false;
+	}
+	struct wlr_xwayland_surface *surface = view->wlr_xwayland_surface;
+	struct sway_xwayland *xwayland = &server.xwayland;
+
+	// TODO: return true if the NET_WM_STATE is MODAL
```
Comment (RyanDwyer) [inline sway/desktop/xwayland.c]: What's the reason for not actioning this todo?

**Comment by SirCmpwn** [comment]
Did we decide it wasn't possible to pull this information out of wlroots?

File: sway/desktop/xwayland.c (hunk 1 above)
Comment (emersion) [reply sway/desktop/xwayland.c]: wlroots doesn't expose this, yet

**Comment by emersion** [comment]
>Did we decide it wasn't possible to pull this information out of wlroots?

Depends what you mean. We could parse these in wlroots, store them in a bitfield in `wlr_xwayland_surface` and read them here instead. But wlroots won't always support all atoms compositors might need.

**Comment by SirCmpwn** [comment]
bleh.

**Comment by emersion** [comment]
Alternatively, we could decide not to expose atoms, and require wlroots to support all atoms a compositor might need.

**Comment by atomnuker** [comment]
Breaks Firefox downloads (clicking on links does nothing). It does work correctly for save windows though.

**Comment by emersion** [comment]
>Breaks Firefox downloads (clicking on links does nothing). It does work correctly for save windows though.

Can you elaborate? The downloads popup and the download section of the library window works for me.

**Comment by emersion** [comment]
>Breaks Firefox downloads (clicking on links does nothing). It does work correctly for save windows though.

So the "Save As" window where you select a destination directory works, but the "Open with… or Save" dialog that appears before is shrunk to 0x0 (with the texture overflowing out of the border).

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**